components/api_server/tests/test_healthcheck.py

```diff
@@ -1,21 +1,48 @@
 """Unit tests for the API-server healthcheck script."""
 
+import sys
 import unittest
 from http import HTTPStatus
 from unittest.mock import patch, MagicMock
 
 
+@patch("sys.exit")
+@patch("urllib.request.urlopen")
 class APIServerHealthcheckTestCase(unittest.TestCase):
     """Unit tests for the API-server healthcheck."""
 
-    @patch("sys.exit")
-    @patch("urllib.request.urlopen")
-    def test_healthy(self, mock_urlopen, mock_exit):
-        """Test that the server is healthy."""
+    def create_response(self, healthy: bool = True, status=HTTPStatus.OK) -> MagicMock:
+        """Create a API-server response."""
         response = MagicMock()
-        response.status = HTTPStatus.OK
+        response.status = status
+        response.read.return_value = b'{"healthy": true}' if healthy else b'{"healthy": false}'
         response.__enter__.return_value = response
-        mock_urlopen.return_value = response
-        import healthcheck  # noqa: F401
+        return response
+
+    def tearDown(self) -> None:
+        """Remove the healthcheck module."""
+        del sys.modules["healthcheck"]
```

Comment:
Is the `del healthcheck` at the end of each test then still needed?

Reply:
Good catch. It isn't needed.